libraries/AC_AutoTune/AC_AutoTune_Multi.cpp

```diff
@@ -0,0 +1,626 @@
+            break;
+        case PITCH:
+            Log_Write_AutoTune(axis, tune_type, target_angle, test_angle_min, test_angle_max, tune_pitch_rp, tune_pitch_rd, tune_pitch_sp, test_accel_max);
+            break;
+        case YAW:
+            Log_Write_AutoTune(axis, tune_type, target_angle, test_angle_min, test_angle_max, tune_yaw_rp, tune_yaw_rLPF, tune_yaw_sp, test_accel_max);
+            break;
+        }
+    } else {
+        switch (axis) {
+        case ROLL:
+            Log_Write_AutoTune(axis, tune_type, target_rate, test_rate_min, test_rate_max, tune_roll_rp, tune_roll_rd, tune_roll_sp, test_accel_max);
+            break;
+        case PITCH:
+            Log_Write_AutoTune(axis, tune_type, target_rate, test_rate_min, test_rate_max, tune_pitch_rp, tune_pitch_rd, tune_pitch_sp, test_accel_max);
+            break;
+        case YAW:
+            Log_Write_AutoTune(axis, tune_type, target_rate, test_rate_min, test_rate_max, tune_yaw_rp, tune_yaw_rLPF, tune_yaw_sp, test_accel_max);
+            break;
+        }
+    }
+
+}
+
+void AC_AutoTune_Multi::Log_AutoTuneDetails()
+{
+    Log_Write_AutoTuneDetails(lean_angle, rotation_rate);
+}
+
+// @LoggerMessage: ATUN
+// @Description: Copter/QuadPlane AutoTune
+// @Vehicles: Copter, Plane
+// @Field: TimeUS: Time since system startup
+// @Field: Axis: which axis is currently being tuned
+// @Field: TuneStep: step in autotune process
+// @Field: Targ: target angle or rate, depending on tuning step
+// @Field: Min: measured minimum target angle or rate
+// @Field: Max: measured maximum target angle or rate
+// @Field: RP: new rate gain P term
+// @Field: RD: new rate gain D term
+// @Field: SP: new angle P term
+// @Field: ddt: maximum measured twitching acceleration
+
+// Write an Autotune data packet
+void AC_AutoTune_Multi::Log_Write_AutoTune(uint8_t _axis, uint8_t tune_step, float meas_target, float meas_min, float meas_max, float new_gain_rp, float new_gain_rd, float new_gain_sp, float new_ddt)
+{
+    AP::logger().Write(
+        "ATUN",
+        "TimeUS,Axis,TuneStep,Targ,Min,Max,RP,RD,SP,ddt",
+        "s--ddd---o",
+        "F--000---0",
+        "QBBfffffff",
+        AP_HAL::micros64(),
+        axis,
+        tune_step,
+        meas_target*0.01f,
+        meas_min*0.01f,
+        meas_max*0.01f,
+        new_gain_rp,
+        new_gain_rd,
+        new_gain_sp,
+        new_ddt);
+}
+
+// Write an Autotune data packet
+void AC_AutoTune_Multi::Log_Write_AutoTuneDetails(float angle_cd, float rate_cds)
+{
+    // @LoggerMessage: ATDE
+    // @Description: AutoTune data packet
+    // @Field: TimeUS: Time since system startup
+    // @Field: Angle: current angle
+    // @Field: Rate: current angular rate
+    AP::logger().Write(
+        "ATDE",
+        "TimeUS,Angle,Rate",
+        "sdk",
+        "F00",
+        "Qff",
+        AP_HAL::micros64(),
+        angle_cd*0.01f,
+        rate_cds*0.01f);
+}
+
+float AC_AutoTune_Multi::get_intra_test_ri()
```

Comment:
This should take an axis argument instead of figuring out which axis is required by checking the "roll_enabled()" etc.  same goes for get_load_tuned_ri()

Reply:
Done.  Thanks!